azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
The intent is to switch between boost::thread/std::thread based on that
define. IIUC std::thread is problematic in DLLs for some stdlib versions
under Windows where boost::thread is not?

On Thu, Apr 2, 2015 at 9:44 AM, Andrey Upadyshev <notifications@github.com>
wrote:

> In azmq/detail/config/thread.hpp
> <https://github.com/zeromq/azmq/pull/71#discussion_r27663189>:
>
> > @@ -0,0 +1,26 @@
> > +/*
> > +    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
> > +
> > +    This file is part of azmq
> > +
> > +    Distributed under the Boost Software License, Version 1.0. (See accompanying
> > +    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
> > +*/
> > +#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +
> > +#if !defined(AZMQ_HAS_STD_THREAD)
>
> Why do we need AZMQ_HAS_STD_THREAD?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/azmq/pull/71/files#r27663189>.
>